### PR TITLE
feat: helia-lan-discovery example

### PR DESCRIPTION
**Pull request by SgtPooki**
[user reached out on slack](https://filecoinproject.slack.com/archives/C046HDAHA13/p1693065664917659) about peer discovery not working quite right with Helia, so i decided to throw together some code and try it out

This example is now working and has some very basic tests

### CODE REVIEW

**Comment by achingbrain** [comment]
Related: https://github.com/libp2p/js-libp2p/issues/2012

**Comment by achingbrain** [comment]
Also related: https://github.com/libp2p/js-libp2p/pull/2013

File: .gitignore [hunk 1]
```diff
@@ -14,3 +14,10 @@ playwright-report
 .parcel-cache
 .envrc
 .tool-versions
+cid.txt
+client-connectedPeers.txt
+client-discoveredPeers.txt
+server-connectedPeers.txt
+server-discoveredPeers.txt
+server-peerId.txt
+client-peerId.txt
```
Comment (SgtPooki) [inline .gitignore:23]:
```suggestion
```

**Review by SgtPooki — COMMENTED** [review]
self review

**Comment by SgtPooki** [comment]
Merged this even though test-monorepo is failing

**Comment by achingbrain** [comment]
> Merged this even though test-monorepo is failing

😞

**Comment by achingbrain** [comment]
This was missing a repo to copy the examples to so it further broke the build.

I've updated the instructions for adding new examples which were a little out of date, I think it's all present & correct now - https://github.com/ipfs-examples/helia-examples#how-to-add-a-new-example

**Comment by achingbrain** [comment]
That said the tests are pretty unreliable, the lan example ones seem to run forever.  There was a `timeout` cli command before the tests, presumably that just kills them if they take too long, which should probably be fixed instead?

**Pull request merged**